cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/preview/PreviewStore.java

```diff
@@ -91,4 +92,30 @@
    */
   @Nullable
   PreviewStatus getPreviewStatus(ApplicationId applicationId);
+
+  /**
+   * Adds the preview request for given application id in waiting state.
+   * @param applicationId the application id corresponding to the request
+   * @param appRequest preview request configuration
+   */
+  void addToWaitingState(ApplicationId applicationId, AppRequest appRequest);
+
+  /**
+   * @return list of all waiting requests in waiting state sorted by submit time
+   */
+  List<PreviewRequest> getAllInWaitingState();
```

Comment:
Why need a special one for waiting state? It should be more generic like `getAllInStatus(PreviewState.Status status)`

Reply:
This is called from PreviewRequestQueue implementation to build the in-memory queue in case manager is restarted. This method is special since it returns the list of `PreviewRequest`.  PreviewRequest is only stored for requests in waiting state.